### PR TITLE
feat(vpn): add data_source_vpn_connection_health_checks

### PR DESCRIPTION
**Pull request by wangyuancheng01**
**What this PR does / why we need it**:add data_source_vpn_connection_health_checks

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note

```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
 root@ecs-wangyuancheng:/home/huawei/go/src/github.com/huaweicloud/terraform-provider-huaweicloud (vpn)$ make testacc TEST="./huaweicloud/services/acceptance/vpn" TESTARGS="-run TestAccVPNConnectionHealthChecksDataSource_Basic"
...
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/vpn -v -run TestAccVPNConnectionHealthChecksDataSource_Basic -timeout 360m -parallel 4
=== RUN   TestAccVPNConnectionHealthChecksDataSource_Basic
=== PAUSE TestAccVPNConnectionHealthChecksDataSource_Basic
=== CONT  TestAccVPNConnectionHealthChecksDataSource_Basic
--- PASS: TestAccVPNConnectionHealthChecksDataSource_Basic (286.07s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/vpn       286.113s
```

### CODE REVIEW

File: docs/data-sources/vpn_connection_health_checks.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `region` - (Optional, String) Specifies the region in which to obtain the VPN connection health check.
+  If omitted, the provider-level region will be used.
+
+* `connection_id` - (Optional, String) Specifies the ID of the VPN connection health check.
```
Comment (Jason-Zhang9309) [inline docs/data-sources/vpn_connection_health_checks.md]:
```suggestion
* `connection_id` - (Optional, String) Specifies the ID of the VPN connection.
```

File: docs/data-sources/vpn_connection_health_checks.md [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+* `connection_id` - The connection ID of the connection health check.
+
+* `type` - The type of the connection health check.
+
+* `source_ip` - The source IP address of the VPN connection.
+
+* `destination_ip` - The destination IP address of the VPN connection.
+
+* `proto_type` - The proto type of the connection health check.
```
Comment (Jason-Zhang9309) [inline docs/data-sources/vpn_connection_health_checks.md]:
```suggestion
## Attribute Reference

In addition to all arguments above, the following attributes are exported:

* `id` -  The  data source ID.

* `connection_health_checks` - All resource connection health checks that match the filter parameters.
  The [connection_health_checks](#connection_health_checks) structure is documented below.

<a name="connection_health_checks"></a>
The `connection_health_checks` block supports:

* `id` -  The ID of the connection health check.

* `status` - The status of the connection health check.

* `connection_id` - The connection ID of the connection health check.

* `type` - The type of the connection health check.

* `source_ip` - The source IP address of the VPN connection.

* `destination_ip` - The destination IP address of the VPN connection.

* `proto_type` - The proto type of the connection health check.
```

File: huaweicloud/services/acceptance/vpn/data_source_huaweicloud_vpn_connection_health_checks_test.go [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+		ProviderFactories: acceptance.TestAccProviderFactories,
+		Steps: []resource.TestStep{
+			{
+				Config: testDataSourceConnectionHealthChecks_basic(rName, ipAddress),
+				Check: resource.ComposeTestCheckFunc(
+					dc.CheckResourceExists(),
+					resource.TestCheckResourceAttrPair(resourceName, "connection_id",
+						"huaweicloud_vpn_connection.test", "id"),
+					resource.TestCheckResourceAttrSet(resourceName, "destination_ip"),
+					resource.TestCheckResourceAttrSet(resourceName, "source_ip"),
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/acceptance/vpn/data_source_huaweicloud_vpn_connection_health_checks_test.go]:
```suggestion
					resource.TestCheckResourceAttrPair(resourceName, "destination_ip", "huaweicloud_vpn_gateway.test", "master_eip.0.id"),
					resource.TestCheckResourceAttrPair(resourceName, "source_ip", "huaweicloud_vpn_customer_gateway.test", "ip"),
```

File: huaweicloud/services/acceptance/vpn/data_source_huaweicloud_vpn_connection_health_checks_test.go [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+					resource.TestCheckResourceAttrSet(resourceName, "status"),
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/acceptance/vpn/data_source_huaweicloud_vpn_connection_health_checks_test.go]:
```suggestion
					resource.TestCheckResourceAttr(resourceName, "status", "ACTIVE"),
```

File: docs/data-sources/vpn_connection_health_checks.md (hunk 1 above)
Comment (wangyuancheng01) [reply docs/data-sources/vpn_connection_health_checks.md]: down

File: docs/data-sources/vpn_connection_health_checks.md (hunk 2 above)
Comment (wangyuancheng01) [reply docs/data-sources/vpn_connection_health_checks.md]: down

File: huaweicloud/services/acceptance/vpn/data_source_huaweicloud_vpn_connection_health_checks_test.go (hunk 3 above)
Comment (wangyuancheng01) [reply huaweicloud/services/acceptance/vpn/data_source_huaweicloud_vpn_connection_health_checks_test.go]: down

File: huaweicloud/services/acceptance/vpn/data_source_huaweicloud_vpn_connection_health_checks_test.go (hunk 4 above)
Comment (wangyuancheng01) [reply huaweicloud/services/acceptance/vpn/data_source_huaweicloud_vpn_connection_health_checks_test.go]: down

**Pull request merged**